std/traits.d

```diff
@@ -4543,6 +4543,41 @@ unittest
     static assert(!__traits(compiles, lvalueOf!byte = 128));
 }
 
+/+
+Given a type $(D T), $(D lvalueInit) represents a unique instance of $(D T)
+initialized to $(D T.init). $(D lvalueInit) is always immutable, regardless
+of $(D T)'s qualifiers.
+
+$(D lvalueInit) is useful when one needs a reference the $(D .init) value,
+without creating duplicate static instances, or temporaries that would require
+destruction.
++/
+package template lvalueInit(T)
+{
+    //static if (is(T == immutable(T)))
+    //    static lvalueInit = T.init;
+    //else
+    //    alias lvalueInit = .lvalueInit!(immutable(T));
+    static lvalueInit = immutable(T).init;
```

Comment:
I think it should be an undefined function, a la:

```d
ref T lvalueof(T)();
```

True it can't be applied everywhere but it's a better template constraint that's impossible to misuse (e.g. write to).